packages/dd-trace/src/tracer.js

```diff
@@ -58,17 +58,25 @@ class DatadogTracer extends Tracer {
     }
   }
 
-  wrap (name, options, fn) {
+  wrap (name, options, fn, requiresParent = false) {
```

Comment:
The existing options object is passed directly to `trace` and is used for `tags`, `childOf`, etc., so that's why I avoided putting it in there. That being said, an option that `trace` doesn't use should be ignored, so I guess I can put it in that object. The other thing to note here is that if the options object is a function returning an object, then that function will have to add the option, but that's fine I guess.

> Any way to express this in a shorter way too? Maybe root or similar?

I **really** like super-descriptive naming. I almost called it `requiresParentSpan`. I don't think `root` conveys enough of what we're instructing the `wrap` function to do here.

Reply:
> The other thing to note here is that if the options object is a function returning an object, then that function will have to add the option, but that's fine I guess.

The side effect here is that now we're running the `options` function unnecessarily, which is bad for performance. A future optimization might be to have the `options` function be a property of the options object, so that we don't have to run it unnecessarily.

Reply:
> That being said, an option that trace doesn't use should be ignored, so I guess I can put it in that object.

It should be used by `tracer.trace` since the functionality should be there.

> I really like super-descriptive naming. I almost called it requiresParentSpan. I don't think root conveys enough of what we're instructing the wrap function to do here.

Fair enough. Since it's called `childOf`, maybe something like `childOfRequired`? Goal being to keep the same terminology.

Reply:
> It should be used by `tracer.trace` since the functionality should be there.

[I explained here why that doesn't necessarily work.](https://github.com/DataDog/dd-trace-js/pull/814#discussion_r369637879)

> Fair enough. Since it's called `childOf`, maybe something like `childOfRequired`? Goal being to keep the same terminology.

I think that's pretty close. The only thing that isn't great here is that it makes it sound like there's something called `required` that this thing is now a child of. Also on second though, `requires` and `required` aren't great since, well, this is Node.js. Maybe `needsChildOf`?

Reply:
`orphanable`? It seems to have more usage in the wild than I expected when I thought of it.

Reply:
Actually I just realized it wouldn't be semantically correct, so scratch that.